### ISSUE TITLE
Undefined symbols for architecture x86_64 - "_crypto_onetimeauth_pick_best_implementation"

### ISSUE THREAD

**Issue by ziogaschr** [jedisct1/libsodium#190]
Hello @jedisct1,
I am trying to use libsodium on xCode 6.0.1 for iOS 8.0.
Unluckily, I am receiving this error during the build of my project.

```
Ld /Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Products/Debug-iphonesimulator/AuthentiQ.app/AuthentiQ normal x86_64
    cd /Users/Destiny/Developer/AuthentiQ
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -arch x86_64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.0.sdk -L/Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Products/Debug-iphonesimulator -L/Users/Destiny/Downloads/libsodium-0.7.1/libsodium-ios/lib -F/Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Products/Debug-iphonesimulator -filelist /Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Intermediates/AuthentiQ.build/Debug-iphonesimulator/AuthentiQ.build/Objects-normal/x86_64/AuthentiQ.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -Xlinker -objc_abi_version -Xlinker 2 -ObjC -ObjC -lPods-AuthentiQ-libsodium -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphonesimulator -Xlinker -add_ast_path -Xlinker /Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Intermediates/AuthentiQ.build/Debug-iphonesimulator/AuthentiQ.build/Objects-normal/x86_64/AuthentiQ.swiftmodule -fobjc-arc -fobjc-link-runtime -Xlinker -no_implicit_dylibs -mios-simulator-version-min=8.0 -lPods-AuthentiQ -Xlinker -dependency_info -Xlinker /Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Intermediates/AuthentiQ.build/Debug-iphonesimulator/AuthentiQ.build/Objects-normal/x86_64/AuthentiQ_dependency_info.dat -o /Users/Destiny/Library/Developer/Xcode/DerivedData/AuthentiQ-agxovwqbnxhsruhcpcdsobmaefst/Build/Products/Debug-iphonesimulator/AuthentiQ.app/AuthentiQ

Undefined symbols for architecture x86_64:
  "_crypto_onetimeauth_pick_best_implementation", referenced from:
      _sodium_init in libPods-AuthentiQ-libsodium.a(core.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

I tried by:
* using cocoapods https://github.com/CocoaPods/Specs/blob/master/Specs/libsodium/0.7.0/libsodium.podspec.json
* create a new cocoapod spec for you latest tag 0.7.1
* build libsodium and add it in the xCode project

Can you please help me?

**Comment by JackWink** [comment]
@chrisballinger do you have any thoughts on [modifying the project via prepare_command](http://stackoverflow.com/questions/20072937/add-run-script-build-phase-to-xcode-project-from-podspec)?

Part of what's bothered me with `prepare_command` is that it's only run on pod installation and not at build-time of the app. It seems like this could fix the slow install issue, but at the same time, I'm not sure how I feel about modifying the project.

**Comment by chrisballinger** [comment]
@JackWink Hey just saw this message. Here's my current version of [libsodium.podspec.json](https://github.com/chrisballinger/BLEMeshChat/blob/master/Podspecs/libsodium.podspec.json) that uses the release .tar.gz (no longer relies on autotools). Compiles the files inside Xcode instead of using the dist-build script, but still needs prepare_command to call configure for generating version.h.

**Comment by WangWenchao** [comment]
Did have fix this bug ？ libsodium-1.0.15 is still met this err
